classes/ffprobe.php

```diff
@@ -141,14 +141,18 @@ public function get_media_metadata(\stored_file $file) : array {
         // ffprobe will not take a stream or a file object.
         // This is not ideal when dealing with massive files or object storage,
         // but there isn't anything that can be done about it.
-        $tempfile = $file->copy_content_to_temp();
+
+        // Write the file contents into a tempfile.
+        $tempfile = tmpfile();
+        fwrite($tempfile, $file->get_content());
```

Comment:
this will put it all into memoru, maybe https://www.php.net/manual/en/function.stream-copy-to-stream.php

Reply:
actually $file->copy_content_to() should be much easier

Reply:
I tested both functions, and they both had a peak memory usage of 24M while churning through 2 1.5Gb files, so I went with $file->copy_content_to as its a little cleaner